src/interfaces/IBPool.sol

```diff
@@ -18,26 +26,85 @@ interface IBPool is IERC20 {
     uint256 tokenAmountOut
   );
 
+  /**
+   * @notice Emitted when a token amount is introduced to the pool
+   * @param caller The caller of the function
+   * @param tokenIn The address of the token being sent to the pool
+   * @param tokenAmountIn The balance of the token being sent to the pool
+   */
   event LOG_JOIN(address indexed caller, address indexed tokenIn, uint256 tokenAmountIn);
 
+  /**
+   * @notice Emitted when a token amount is removed from the pool
+   * @param caller The caller of the function
+   * @param tokenOut The address of the token being removed from the pool
+   * @param tokenAmountOut The balance of the token being removed from the pool
+   */
   event LOG_EXIT(address indexed caller, address indexed tokenOut, uint256 tokenAmountOut);
 
+  /**
+   * @notice Emitted when a call is performed to the pool
+   * @param sig The signature of the function selector being called
+   * @param caller The caller of the function
+   * @param data The complete data of the call
+   */
   event LOG_CALL(bytes4 indexed sig, address indexed caller, bytes data) anonymous;
 
+  /**
+   * @notice Sets the new swap fee
+   * @param swapFee The new swap fee
+   */
   function setSwapFee(uint256 swapFee) external;
 
+  /**
+   * @notice Sets the new controller
+   * @param manager The new controller
+   */
   function setController(address manager) external;
 
+  /**
+   * @notice Finalize the pool, removing the restrictions on the pool
+   */
   function finalize() external;
 
+  /**
+   * @notice Binds a token to the pool
+   * @param token The address of the token to bind
+   * @param balance The balance of the token to bind
+   * @param denorm The denormalized weight of the token to bind
+   */
   function bind(address token, uint256 balance, uint256 denorm) external;
 
+  /**
+   * @notice Unbinds a token from the pool
+   * @param token The address of the token to unbind
+   */
   function unbind(address token) external;
 
+  /**
+   * @notice Joins a pool, providing each token in the pool with a proportional amount
+   * @param poolAmountOut The amount of pool tokens to mint
+   * @param maxAmountsIn The maximum amount of tokens to send to the pool
+   */
   function joinPool(uint256 poolAmountOut, uint256[] calldata maxAmountsIn) external;
 
+  /**
+   * @notice Exits a pool, receiving each token in the pool with a proportional amount
+   * @param poolAmountIn The amount of pool tokens to burn
+   * @param minAmountsOut The minimum amount of tokens to receive from the pool
+   */
   function exitPool(uint256 poolAmountIn, uint256[] calldata minAmountsOut) external;
 
+  /**
+   * @notice Swaps an exact amount of tokens in for as many tokens out as possible
```

Comment:
*for a calculated amount of token out

Reply:
Wdym? When I say **_as many tokens out as possible_** I mean you not expect a specific amount. For me **_"or a calculated amount of token out"_** means that the user previously calculated the amount out, which is not exactly the case here.

Reply:
i'd say an amount of tokens out, the "as many as possible" isn't actually something..